# Data Audit
## Understanding the dataset before cleaning or analysis

**Goal:** Load all datasets and understand:
- Shape & structure
- Columns & data types
- Missing values & duplicates
- Data relationships
- Value ranges & distributions

**Output:** Problems list for cleaning step

<hr>

## Import Libraries

In [1]:
import numpy as np
import pandas as pd

<hr>

## Load All Datasets

In [2]:
customers = pd.read_csv(r"D:\Ayush\Learning\Projects\Ecommerce\data\raw\olist_customers_dataset.csv")
geolocation = pd.read_csv(r"D:\Ayush\Learning\Projects\Ecommerce\data\raw\olist_geolocation_dataset.csv")
order_items = pd.read_csv(r"D:\Ayush\Learning\Projects\Ecommerce\data\raw\olist_order_items_dataset.csv")
payments = pd.read_csv(r"D:\Ayush\Learning\Projects\Ecommerce\data\raw\olist_order_payments_dataset.csv")
reviews = pd.read_csv(r"D:\Ayush\Learning\Projects\Ecommerce\data\raw\olist_order_reviews_dataset.csv")
orders = pd.read_csv(r"D:\Ayush\Learning\Projects\Ecommerce\data\raw\olist_orders_dataset.csv")
products = pd.read_csv(r"D:\Ayush\Learning\Projects\Ecommerce\data\raw\olist_products_dataset.csv")
sellers = pd.read_csv(r"D:\Ayush\Learning\Projects\Ecommerce\data\raw\olist_sellers_dataset.csv")
category_translation = pd.read_csv(r"D:\Ayush\Learning\Projects\Ecommerce\data\raw\product_category_name_translation.csv")

print("***All datasets loaded successfully!***")

***All datasets loaded successfully!***


<hr>

## Check Size of All Datasets

In [3]:
print("*"*10,"DATASET SIZES","*"*10)
print()
print(f"Customers:              {customers.shape}")
print(f"Geolocation:            {geolocation.shape}")
print(f"Order Items:            {order_items.shape}")
print(f"Payments:               {payments.shape}")
print(f"Reviews:                {reviews.shape}")
print(f"Orders:                 {orders.shape}")
print(f"Products:               {products.shape}")
print(f"Sellers:                {sellers.shape}")
print(f"Category Translation:   {category_translation.shape}")
print("\n" + "-" * 50)
print(f"TOTAL ROWS: {customers.shape[0] + geolocation.shape[0] + order_items.shape[0] + payments.shape[0] + reviews.shape[0] + orders.shape[0] + products.shape[0] + sellers.shape[0] + category_translation.shape[0]:,}")

********** DATASET SIZES **********

Customers:              (99441, 5)
Geolocation:            (1000163, 5)
Order Items:            (112650, 7)
Payments:               (103886, 5)
Reviews:                (99224, 7)
Orders:                 (99441, 8)
Products:               (32951, 9)
Sellers:                (3095, 4)
Category Translation:   (71, 2)

--------------------------------------------------
TOTAL ROWS: 1,550,922


<hr>

## Column Names & Data Types for Each Dataset

### Customers Table

In [4]:
print("*"*10,"CUSTOMERS TABLE","*"*10)
print("\nColumn Names & Types:")
customers.dtypes

********** CUSTOMERS TABLE **********

Column Names & Types:


customer_id                   str
customer_unique_id            str
customer_zip_code_prefix    int64
customer_city                 str
customer_state                str
dtype: object

In [5]:
print("First few rows:")
customers.head()

First few rows:


,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP
1,18955e83d337fd6b2def6b18a428ac77,290c77bc529b7ac935b93aa66c333dc3,9790,sao bernardo do campo,SP
2,4e7b3e00288586ebd08712fdd0374a03,060e732b5b29e8181a18229c7b0b2b5e,1151,sao paulo,SP
3,b2b6027bc5c5109e529d4dc6358b12c3,259dac757896d24d7702b9acbbff3f3c,8775,mogi das cruzes,SP
4,4f2d8ab171c80ec8364f7c12e35b23ad,345ecd01c38d18a9036ed96c73b8d066,13056,campinas,SP


### Orders Table

In [6]:
print("*"*10,"ORDERS TABLE","*"*10)
print("\nColumn Names & Types:")
orders.dtypes

********** ORDERS TABLE **********

Column Names & Types:


order_id                         str
customer_id                      str
order_status                     str
order_purchase_timestamp         str
order_approved_at                str
order_delivered_carrier_date     str
order_delivered_customer_date    str
order_estimated_delivery_date    str
dtype: object

In [7]:
print("First few rows:")
orders.head()

First few rows:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00
3,949d5b44dbf5de918fe9c16f97b45f8a,f88197465ea7920adcdbec7375364d82,delivered,2017-11-18 19:28:06,2017-11-18 19:45:59,2017-11-22 13:39:59,2017-12-02 00:28:42,2017-12-15 00:00:00
4,ad21c59c0840e6cb83a9ceb5573f8159,8ab97904e6daea8866dbdbc4fb7aad2c,delivered,2018-02-13 21:18:39,2018-02-13 22:20:29,2018-02-14 19:46:34,2018-02-16 18:17:02,2018-02-26 00:00:00


### Order Items Table

In [8]:
print("*"*10,"ORDER ITEMS TABLE","*"*10)
print("\nColumn Names & Types:")
order_items.dtypes

********** ORDER ITEMS TABLE **********

Column Names & Types:


order_id                   str
order_item_id            int64
product_id                 str
seller_id                  str
shipping_limit_date        str
price                  float64
freight_value          float64
dtype: object

In [9]:
print("First few rows:")
order_items.head()

First few rows:


,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.90,13.29
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,2017-05-03 11:05:13,239.90,19.93
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,2018-01-18 14:48:30,199.00,17.87
3,00024acbcdf0a6daa1e931b038114c75,1,7634da152a4610f1595efa32f14722fc,9d7a1d34a5052409006425275ba1c2b4,2018-08-15 10:10:18,12.99,12.79
4,00042b26cf59d7ce69dfabb4e55b4fd9,1,ac6c3623068f30de03045865e4e10089,df560393f3a51e74553ab94004ba5c87,2017-02-13 13:57:51,199.90,18.14


### Payments Table

In [10]:
print("*"*10,"PAYMENTS TABLE","*"*10)
print("\nColumn Names & Types:")
payments.dtypes

********** PAYMENTS TABLE **********

Column Names & Types:


order_id                    str
payment_sequential        int64
payment_type                str
payment_installments      int64
payment_value           float64
dtype: object

In [11]:
print("First few rows: ")
payments.head()

First few rows: 


,order_id,payment_sequential,payment_type,payment_installments,payment_value
0,b81ef226f3fe1789b1e8b2acac839d17,1,credit_card,8,99.33
1,a9810da82917af2d9aefd1278f1dcfa0,1,credit_card,1,24.39
2,25e8ea4e93396b6fa0d3dd708e76c1bd,1,credit_card,1,65.71
3,ba78997921bbcdc1373bb41e913ab953,1,credit_card,8,107.78
4,42fdf880ba16b47b59251dd489d4441a,1,credit_card,2,128.45


### Reviews Table

In [12]:
print("*"*10,"REVIEWS TABLE","*"*10)
print("\nColumn Names & Types:")
reviews.dtypes

********** REVIEWS TABLE **********

Column Names & Types:


review_id                    str
order_id                     str
review_score               int64
review_comment_title         str
review_comment_message       str
review_creation_date         str
review_answer_timestamp      str
dtype: object

In [13]:
print("First few rows: ")
reviews.head()

First few rows: 


,review_id,order_id,review_score,review_comment_title,review_comment_message,review_creation_date,review_answer_timestamp
0,7bc2406110b926393aa56f80a40eba40,73fc7af87114b39712e6da79b0a377eb,4,NaN,NaN,2018-01-18 00:00:00,2018-01-18 21:46:59
1,80e641a11e56f04c1ad469d5645fdfde,a548910a1c6147796b98fdf73dbeba33,5,NaN,NaN,2018-03-10 00:00:00,2018-03-11 03:05:13
2,228ce5500dc1d8e020d8d1322874b6f0,f9e4b658b201a9f2ecdecbb34bed034b,5,NaN,NaN,2018-02-17 00:00:00,2018-02-18 14:36:24
3,e64fb393e7b32834bb789ff8bb30750e,658677c97b385a9be170737859d3511b,5,NaN,Recebi bem antes do prazo estipulado.,2017-04-21 00:00:00,2017-04-21 22:02:06
4,f7c4243c7fe1938f181bec41a392bdeb,8e6bfb81e283fa7e4f11123a3fb894f1,5,NaN,Parabéns lojas lannister adorei comprar pela I...,2018-03-01 00:00:00,2018-03-02 10:26:53


### Products Table

In [14]:
print("*"*10,"PRODUCTS TABLE","*"*10)
print("\nColumn Names & Types:")
products.dtypes

********** PRODUCTS TABLE **********

Column Names & Types:


product_id                        str
product_category_name             str
product_name_lenght           float64
product_description_lenght    float64
product_photos_qty            float64
product_weight_g              float64
product_length_cm             float64
product_height_cm             float64
product_width_cm              float64
dtype: object

In [15]:
print("First few rows: ")
products.head()

First few rows: 


,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.0,287.0,1.0,225.0,16.0,10.0,14.0
1,3aa071139cb16b67ca9e5dea641aaa2f,artes,44.0,276.0,1.0,1000.0,30.0,18.0,20.0
2,96bd76ec8810374ed1b65e291975717f,esporte_lazer,46.0,250.0,1.0,154.0,18.0,9.0,15.0
3,cef67bcfe19066a932b7673e239eb23d,bebes,27.0,261.0,1.0,371.0,26.0,4.0,26.0
4,9dc1a7de274444849c219cff195d0b71,utilidades_domesticas,37.0,402.0,4.0,625.0,20.0,17.0,13.0


### Sellers Table

In [16]:
print("*"*10,"SELLERS TABLE","*"*10)
print("\nColumn Names & Types:")
sellers.dtypes

********** SELLERS TABLE **********

Column Names & Types:


seller_id                   str
seller_zip_code_prefix    int64
seller_city                 str
seller_state                str
dtype: object

In [17]:
print("First few rows:")
sellers.head()

First few rows:


,seller_id,seller_zip_code_prefix,seller_city,seller_state
0,3442f8959a84dea7ee197c632cb2df15,13023,campinas,SP
1,d1b65fc7debc3361ea86b5f14c68d2e2,13844,mogi guacu,SP
2,ce3ad9de960102d0677a81f5d0bb7b2d,20031,rio de janeiro,RJ
3,c0f3eea2e14555b6faeea3dd58c1b1c3,4195,sao paulo,SP
4,51a04a8a6bdcb23deccc82b0b80742cf,12914,braganca paulista,SP


### Geolocation Table

In [18]:
print("*"*10,"GEOLOCATION TABLE","*"*10)
print("\nColumn Names & Types:")
geolocation.dtypes

********** GEOLOCATION TABLE **********

Column Names & Types:


geolocation_zip_code_prefix      int64
geolocation_lat                float64
geolocation_lng                float64
geolocation_city                   str
geolocation_state                  str
dtype: object

In [19]:
print("First few rows:")
geolocation.head()

First few rows:


,geolocation_zip_code_prefix,geolocation_lat,geolocation_lng,geolocation_city,geolocation_state
0,1037,-23.545621,-46.639292,sao paulo,SP
1,1046,-23.546081,-46.644820,sao paulo,SP
2,1046,-23.546129,-46.642951,sao paulo,SP
3,1041,-23.544392,-46.639499,sao paulo,SP
4,1035,-23.541578,-46.641607,sao paulo,SP


### Category Translation Table

In [20]:
print("*"*10,"CATEGORY TRANSLATION TABLE","*"*10)
print("\nColumn Names & Types:")
category_translation.dtypes

********** CATEGORY TRANSLATION TABLE **********

Column Names & Types:


product_category_name            str
product_category_name_english    str
dtype: object

In [21]:
print("\nFirst 10 rows:")
category_translation.head(10)


First 10 rows:


,product_category_name,product_category_name_english
0,beleza_saude,health_beauty
1,informatica_acessorios,computers_accessories
2,automotivo,auto
3,cama_mesa_banho,bed_bath_table
4,moveis_decoracao,furniture_decor
5,esporte_lazer,sports_leisure
6,perfumaria,perfumery
7,utilidades_domesticas,housewares
8,telefonia,telephony
9,relogios_presentes,watches_gifts


<hr>

## Missing Values Check

In [22]:
def check_missing_values(df, name):
    print(f"\n{name}")
    print("-" * 60)
    
    missing = df.isnull().sum()
    missing_percent = (df.isnull().sum() / len(df)) * 100
    
    missing_df = pd.DataFrame({
        'Column': df.columns,
        'Missing_Count': missing.values,
        'Missing_Percent': missing_percent.values
    })
    
    missing_with_nulls = missing_df[missing_df['Missing_Count'] > 0]
    
    if len(missing_with_nulls) == 0:
        print("--> No missing values")
    else:
        print(missing_with_nulls.to_string(index=False))

check_missing_values(customers, "CUSTOMERS")
check_missing_values(orders, "ORDERS")
check_missing_values(order_items, "ORDER ITEMS")
check_missing_values(payments, "PAYMENTS")
check_missing_values(reviews, "REVIEWS")
check_missing_values(products, "PRODUCTS")
check_missing_values(sellers, "SELLERS")
check_missing_values(geolocation, "GEOLOCATION")
check_missing_values(category_translation, "CATEGORY TRANSLATION")


CUSTOMERS
------------------------------------------------------------
--> No missing values

ORDERS
------------------------------------------------------------
                       Column  Missing_Count  Missing_Percent
            order_approved_at            160         0.160899
 order_delivered_carrier_date           1783         1.793023
order_delivered_customer_date           2965         2.981668

ORDER ITEMS
------------------------------------------------------------
--> No missing values

PAYMENTS
------------------------------------------------------------
--> No missing values

REVIEWS
------------------------------------------------------------
                Column  Missing_Count  Missing_Percent
  review_comment_title          87656        88.341530
review_comment_message          58247        58.702532

PRODUCTS
------------------------------------------------------------
                    Column  Missing_Count  Missing_Percent
     product_category_name         

<hr>

## Duplicates Check

In [23]:
def check_duplicates(df, name, key_column):
    print(f"\n{name}")
    print("-" * 60)
    
    total_duplicates = df.duplicated().sum()
    print(f"Total duplicate rows (entire row): {total_duplicates}")
    
    if key_column:
        key_duplicates = df[key_column].duplicated().sum()
        print(f"Duplicate {key_column}: {key_duplicates}")

check_duplicates(customers, "CUSTOMERS", 'customer_id')
check_duplicates(orders, "ORDERS", 'order_id')
check_duplicates(order_items, "ORDER ITEMS", 'order_id') 
check_duplicates(payments, "PAYMENTS", 'order_id')  
check_duplicates(reviews, "REVIEWS", 'review_id')
check_duplicates(products, "PRODUCTS", 'product_id')
check_duplicates(sellers, "SELLERS", 'seller_id')
check_duplicates(geolocation, "GEOLOCATION", 'geolocation_zip_code_prefix')
check_duplicates(category_translation, "CATEGORY TRANSLATION", 'product_category_name')


CUSTOMERS
------------------------------------------------------------
Total duplicate rows (entire row): 0
Duplicate customer_id: 0

ORDERS
------------------------------------------------------------
Total duplicate rows (entire row): 0
Duplicate order_id: 0

ORDER ITEMS
------------------------------------------------------------
Total duplicate rows (entire row): 0
Duplicate order_id: 13984

PAYMENTS
------------------------------------------------------------
Total duplicate rows (entire row): 0
Duplicate order_id: 4446

REVIEWS
------------------------------------------------------------
Total duplicate rows (entire row): 0
Duplicate review_id: 814

PRODUCTS
------------------------------------------------------------
Total duplicate rows (entire row): 0
Duplicate product_id: 0

SELLERS
------------------------------------------------------------
Total duplicate rows (entire row): 0
Duplicate seller_id: 0

GEOLOCATION
------------------------------------------------------------


<br>
Duplicate rows were checked across all datasets.

- Customers, orders, products, sellers, and category translation datasets do not have duplicate IDs, which is expected for primary keys.

- order_items and payments contain duplicate order_id values. This is normal because:
  - One order can have multiple items
  - One order can have multiple payment records

- Geolocation dataset contains a large number of duplicate rows and repeated zip code prefixes. This is expected since multiple entries can exist for the same location.

- Reviews dataset has duplicate review_id values. This is not expected and may indicate data inconsistency or duplicate entries or Maybe it's allow in this dataset that one person can give multiple review . This will be investigated further during the cleaning phase.

Overall, most duplicates are expected based on dataset structure, except for the review_id duplicates which need further checking.

<hr>

## Categorical Values Distribution

### Order Status Distribution

In [24]:
print("ORDER STATUS DISTRIBUTION")
print("-" * 60)
status_counts = orders['order_status'].value_counts()
print(status_counts)
print(f"\nUnique statuses: {orders['order_status'].nunique()}")

ORDER STATUS DISTRIBUTION
------------------------------------------------------------
order_status
delivered      96478
shipped         1107
canceled         625
unavailable      609
invoiced         314
processing       301
created            5
approved           2
Name: count, dtype: int64

Unique statuses: 8


### Payment Type Distribution

In [25]:
print("PAYMENT TYPE DISTRIBUTION")
print("-" * 60)
payment_counts = payments['payment_type'].value_counts()
print(payment_counts)
print(f"\nUnique payment types: {payments['payment_type'].nunique()}")

PAYMENT TYPE DISTRIBUTION
------------------------------------------------------------
payment_type
credit_card    76795
boleto         19784
voucher         5775
debit_card      1529
not_defined        3
Name: count, dtype: int64

Unique payment types: 5


### Review Score Distribution

In [26]:
print("REVIEW SCORE DISTRIBUTION")
print("-" * 60)
review_counts = reviews['review_score'].value_counts().sort_index()
print(review_counts)
print(f"\nAverage review score: {reviews['review_score'].mean():.2f}")

REVIEW SCORE DISTRIBUTION
------------------------------------------------------------
review_score
1    11424
2     3151
3     8179
4    19142
5    57328
Name: count, dtype: int64

Average review score: 4.09


### Product Category Distribution

In [27]:
print("TOP 20 PRODUCT CATEGORIES")
print("-" * 60)
category_counts = products['product_category_name'].value_counts().head(20)
print(category_counts)
print(f"\nTotal unique categories: {products['product_category_name'].nunique()}")

TOP 20 PRODUCT CATEGORIES
------------------------------------------------------------
product_category_name
cama_mesa_banho                      3029
esporte_lazer                        2867
moveis_decoracao                     2657
beleza_saude                         2444
utilidades_domesticas                2335
automotivo                           1900
informatica_acessorios               1639
brinquedos                           1411
relogios_presentes                   1329
telefonia                            1134
bebes                                 919
perfumaria                            868
papelaria                             849
fashion_bolsas_e_acessorios           849
cool_stuff                            789
ferramentas_jardim                    753
pet_shop                              719
eletronicos                           517
construcao_ferramentas_construcao     400
eletrodomesticos                      370
Name: count, dtype: int64

Total unique categories:

### State Distribution (Customers)

In [28]:
print("CUSTOMER DISTRIBUTION BY STATE")
print("-" * 60)
state_counts = customers['customer_state'].value_counts()
print(state_counts)
print(f"\nTotal states: {customers['customer_state'].nunique()}")

CUSTOMER DISTRIBUTION BY STATE
------------------------------------------------------------
customer_state
SP    41746
RJ    12852
MG    11635
RS     5466
PR     5045
SC     3637
BA     3380
DF     2140
ES     2033
GO     2020
PE     1652
CE     1336
PA      975
MT      907
MA      747
MS      715
PB      536
PI      495
RN      485
AL      413
SE      350
TO      280
RO      253
AM      148
AC       81
AP       68
RR       46
Name: count, dtype: int64

Total states: 27


<hr>

## Numeric Data Statistics

### Order Items Price & Freight

In [29]:
print("ORDER ITEMS - PRICE STATISTICS")
print("-" * 60)
print("\nPrice (product price):\n")
print(order_items['price'].describe())
print("-" * 60)
print(f"Min price: {order_items['price'].min()}")
print(f"Max price: {order_items['price'].max()}")
print(f"Prices is zero: {(order_items['price'] == 0).sum()}")

ORDER ITEMS - PRICE STATISTICS
------------------------------------------------------------

Price (product price):

count    112650.000000
mean        120.653739
std         183.633928
min           0.850000
25%          39.900000
50%          74.990000
75%         134.900000
max        6735.000000
Name: price, dtype: float64
------------------------------------------------------------
Min price: 0.85
Max price: 6735.0
Prices is zero: 0


### Order Price Insights

* The average price is around 120, but the median is around 75.
  This means most products are actually on the cheaper side.

* Most of the products fall between 40 and 135, so the platform mainly sells affordable items.

* There are some very expensive products (up to 6735), but they are very few.

* There are no products with price = 0, which means the pricing data looks clean.

**Simple understanding:**
Most people are buying low to mid-priced products, and only a few expensive items exist.

<br>

In [30]:
print("ORDER ITEMS - FREIGHT STATISTICS")
print("-" * 60)
print("Freight Value:\n")
print(order_items['freight_value'].describe())
print("-" * 60)
print(f"Min freight: {order_items['freight_value'].min()}")
print(f"Max freight: {order_items['freight_value'].max()}")

ORDER ITEMS - FREIGHT STATISTICS
------------------------------------------------------------
Freight Value:

count    112650.000000
mean         19.990320
std          15.806405
min           0.000000
25%          13.080000
50%          16.260000
75%          21.150000
max         409.680000
Name: freight_value, dtype: float64
------------------------------------------------------------
Min freight: 0.0
Max freight: 409.68


### Freight Value Insight

* Freight value is the shipping cost for each order.

* It depends on distance, product size, and delivery type.

**Simple understanding:**
Higher freight means higher delivery cost, which can affect customer decisions.

**Next step:**
I can check if high freight leads to bad reviews or fewer purchases.

<br>

### Payment Amount Statistics

In [31]:
print("PAYMENTS - AMOUNT STATISTICS")
print("-" * 60)
print(payments['payment_value'].describe())
print("-" * 60)
print(f"\nMin payment: {payments['payment_value'].min()}")
print(f"Max payment: {payments['payment_value'].max()}")
print(f"Payments is zero: {(payments['payment_value'] == 0).sum()}")

PAYMENTS - AMOUNT STATISTICS
------------------------------------------------------------
count    103886.000000
mean        154.100380
std         217.494064
min           0.000000
25%          56.790000
50%         100.000000
75%         171.837500
max       13664.080000
Name: payment_value, dtype: float64
------------------------------------------------------------

Min payment: 0.0
Max payment: 13664.08
Payments is zero: 9


### Payment Insights

* The average payment is around 154, while the median is 100.
  This shows that some high-value orders are increasing the average.

* Most payments are in a normal range, but a few go very high (up to 13664).

* I found 9 records where payment value is 0, which should not happen in normal cases and product prices do not have any zero values, but payments have 9 zero values.

**Possible reasons:**

* order got cancelled
* payment failed
* or data issue

**What I will do:**
I will remove or modiefy these records as per the needs because they can affect analysis.

**Simple understanding:**
Most customers spend a normal amount, but a few big orders increase overall revenue.

<br>

### Product Dimensions & Weight

In [32]:
print("PRODUCTS - DIMENSIONS & WEIGHT STATISTICS")
print("-" * 60)
print("\nProduct Weight (grams): \n")
print(products['product_weight_g'].describe())
print("-" * 60)
print("\nProduct Length (cm):\n")
print(products['product_length_cm'].describe())
print("-" * 60)
print("\nProduct Height (cm):\n")
print(products['product_height_cm'].describe())
print("-" * 60)
print("\nProduct Width (cm):\n")
print(products['product_width_cm'].describe())

PRODUCTS - DIMENSIONS & WEIGHT STATISTICS
------------------------------------------------------------

Product Weight (grams): 

count    32949.000000
mean      2276.472488
std       4282.038731
min          0.000000
25%        300.000000
50%        700.000000
75%       1900.000000
max      40425.000000
Name: product_weight_g, dtype: float64
------------------------------------------------------------

Product Length (cm):

count    32949.000000
mean        30.815078
std         16.914458
min          7.000000
25%         18.000000
50%         25.000000
75%         38.000000
max        105.000000
Name: product_length_cm, dtype: float64
------------------------------------------------------------

Product Height (cm):

count    32949.000000
mean        16.937661
std         13.637554
min          2.000000
25%          8.000000
50%         13.000000
75%         21.000000
max        105.000000
Name: product_height_cm, dtype: float64
-------------------------------------------------------